presto-jwt-authenticators/src/main/java/com/facebook/presto/jwt/Default/DefaultJWTAuthenticator.java

```diff
@@ -0,0 +1,249 @@
+import static com.google.common.io.Files.asCharSource;
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.US_ASCII;
+import static java.util.Base64.getMimeDecoder;
+import static java.util.Objects.requireNonNull;
+
+public class DefaultJWTAuthenticator
+        implements JWTAuthenticator
+{
+    private final JwtParser jwtParser;
+    private final Function<JwsHeader<?>, Key> keyLoader;
+    private static final String KEY_ID_VARIABLE = "${KID}";
+    private HttpServletRequest request;
+    private static final String DEFAULT_KEY = "default-key";
+    private static final CharMatcher INVALID_KID_CHARS = inRange('a', 'z').or(inRange('A', 'Z')).or(inRange('0', '9')).or(CharMatcher.anyOf("_-")).negate();
+
+    public DefaultJWTAuthenticator(JsonWebTokenConfig config)
+    {
+        requireNonNull(config, "config is null");
+        if (config.getKeyFile().contains(KEY_ID_VARIABLE)) {
+            keyLoader = new DynamicKeyLoader(config.getKeyFile());
+        }
+        else {
+            keyLoader = new StaticKeyLoader(config.getKeyFile());
+        }
+        JwtParser jwtParser = Jwts.parserBuilder()
+                .deserializeJsonWith(new JacksonDeserializer<>(ImmutableMap.of(AUTHORIZED_IDENTITY_ATTRIBUTE, AuthorizedIdentity.class)))
+                .setSigningKeyResolver(new SigningKeyResolver()
+                {
+                    // interface uses raw types and this can not be fixed here
+                    @SuppressWarnings("rawtypes")
+                    @Override
+                    public Key resolveSigningKey(JwsHeader header, Claims claims)
+                    {
+                        return keyLoader.apply(header);
+                    }
+
+                    @SuppressWarnings("rawtypes")
+                    @Override
+                    public Key resolveSigningKey(JwsHeader header, String plaintext)
+                    {
+                        return keyLoader.apply(header);
+                    }
+                })
+                .build();
+
+        if (config.getRequiredIssuer() != null) {
+            jwtParser.requireIssuer(config.getRequiredIssuer());
+        }
+        if (config.getRequiredAudience() != null) {
+            jwtParser.requireAudience(config.getRequiredAudience());
+        }
+        this.jwtParser = jwtParser;
+    }
+
+    private static AuthenticationException needAuthentication(String message)
+    {
+        return new AuthenticationException(message, "Bearer realm=\"Presto\", token_type=\"JWT\"");
+    }
+    @Override
+    public Principal createAuthenticatedPrincipal(String token)
+    {
+        try {
+            Jws<Claims> claimsJws = jwtParser.parseClaimsJws(token);
+
+            AuthorizedIdentity authorizedIdentity = claimsJws.getBody().get(AUTHORIZED_IDENTITY_ATTRIBUTE, AuthorizedIdentity.class);
+            if (authorizedIdentity != null) {
+                setAuthorizedIdentity(request, authorizedIdentity);
```

Comment:
We don't have the original request in this class, it is null.

Due to the recent changes added in `JsonWebTokenAuthenticator` for parsing and setting `authorizedIdentity,` we will have to change the new `JWTAuthenticator` interface and add `HttpServletRequest request` in the `createAuthenticatedPrincipal` method arguments.